### PR TITLE
Refactor/pm 20-207/club selection

### PR DESCRIPTION
**Pull request by mati-1**
[PM20-207](https://playmakerpro.atlassian.net/browse/PM20-207)

### CODE REVIEW

File: components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    show={openModal}
+    closeModal={closeModal}
+    contentClassName="h-screen"
+    noContentPaddingX
+  >
+    {values?.map(({ id, name, isActualSeason }) => (
+      <SelectItem
+        key={id}
+        text={name}
+        additionalText={isActualSeason ? 'Aktualny sezon rozgrywkowy' : ''}
```
Comment (dejwi) [inline components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx:25]: It's competitions modal 🌝

File: components/login-register/common/NothingMessage/nothing-message.tsx [hunk 2]
```diff
@@ -1,27 +1,17 @@
...
     </p>
-
-    <Button
-      onClick={onClear}
-      intent="secondary"
-      size="extraSmall"
-      desktopSize="small"
-      text="Wyczyść"
-    />
   </motion.div>
```
Comment (dejwi) [inline components/login-register/common/NothingMessage/nothing-message.tsx:16]: nooo they removed clear btn only for clubs modal, it's still there in previous stuff

Mybe make the prop onClear optional and only render it when it's passed

Comment (dejwi) [inline components/login-register/common/NothingMessage/nothing-message.tsx:16]: nooo they removed clear btn only for clubs modal, it's still there in previous stuff

Mybe make the prop onClear optional and only render it when it's passed

**Review by dejwi — CHANGES_REQUESTED** [review]
🌝🌚🌝🌚🌝🌚🌝🌚🌝🌚🌝🌚🌝

File: components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+        selected={selected.includes(id)}
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx:26]: it works fine bro

File: components/login-register/common/Input/select-input.tsx [hunk 4]
```diff
@@ -1,47 +1,91 @@
...
 import ArrowDownIcon from '@/icons/login/ArrowDown.svg'
 
 interface IProps {
   title: string
   type: string
   inputValue: string[] | string
   setOpenModal: () => void
+  isOpenMenu?: boolean
+  onSave?: (id: string) => void
+  onClose?: () => void
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/common/Input/select-input.tsx:17]: look at the base data form in that case opening a new modal with onclick on the selected input, onSave n onClose isnt necessary

File: components/login-register/register/pages/adding-club-page/adding-club-form.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  const { data: seasonsList } = useSeasonsList()
+  const [openClubsModal, setOpenClubsModal] = useState(false)
+  const [openSeasonModal, setOpenSeasonModal] = useState(false)
+
+  const filteredTeam = clubsList
+    .flatMap(({ teams }) => teams)
+    .filter(({ id }) => selectedClub.includes(id))[0]
+
+  const filteredSeason = seasonsList?.find(({ id }) =>
+    selectedSeason.includes(id),
```
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/adding-club-page/adding-club-form.tsx:31]: it works fine bro

File: components/login-register/common/NothingMessage/nothing-message.tsx (hunk 2 above)
Comment (mati-1) [reply components/login-register/common/NothingMessage/nothing-message.tsx:16]: goddamn

File: components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (dejwi) [reply components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx:26]: but don't do that, why would you switch to that right now from equals signs ? At first I thought that was an array type and I was confused

please change it

File: components/login-register/common/Input/select-input.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (dejwi) [reply components/login-register/common/Input/select-input.tsx:17]: hmmmmm okay

File: components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx (hunk 1 above)
Comment (mati-1) [reply components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx:25]: 😵‍💫 😵‍💫 😵‍💫 😵‍💫 😵‍💫

File: components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (mati-1) [reply components/login-register/register/pages/adding-club-page/modals/competitions-modal.tsx:26]: sure

File: components/login-register/common/NothingMessage/nothing-message.tsx [hunk 6]
```diff
@@ -13,5 +14,14 @@ export const NothingMessage = () => (
     <p className="max-w-[15rem] text-center font-inter text-additional text-body-sm sd:text-body-md">
       Brak wyników dla wpisanej frazy
```
Comment (dejwi) [inline components/login-register/common/NothingMessage/nothing-message.tsx:15]: and please don't forget restore the text that is with clear btn

File: components/common/Modal/header.tsx [hunk 7]
```diff
@@ -26,28 +32,36 @@ const ModalHeader = ({
   setSearchValue,
   withClosingArrow = false,
   withClosingCross = true,
+  searchInputInHeaderTop,
 }: IProps) => (
   <div
     className={clsx(
-      'relative z-20 box-content flex flex-col py-7 px-6 sd:px-8',
+      'relative z-20 box-content flex flex-col px-6 pt-7 !pb-6 sd:px-8',
```
Comment (dejwi) [inline components/common/Modal/header.tsx:39]: no bro this changes all modals and now padding are wrong on normal modals and there's no variant of modal that has different pt and pb

and do we need `searchInputInHeaderTop` prop ? can't we guess that based on presence of the title ?

File: components/login-register/register/pages/adding-club-page/modals/clubs-modal.tsx [hunk 8]
```diff
@@ -110,7 +108,7 @@ export const ClubsModal = ({
         modalTitle="Dodaj klub"
```
Comment (dejwi) [inline components/login-register/register/pages/adding-club-page/modals/clubs-modal.tsx:108]: this modal has an arrow too

**Review by dejwi — CHANGES_REQUESTED** [review]
bro get these paddings right I know you can do it match the figma, check every modal

compare px in padding using computed tab in inspect element on browser

File: components/common/Modal/header.tsx (hunk 7 above)
Comment (mati-1) [reply components/common/Modal/header.tsx:39]: good

**Review by dejwi — APPROVED** [review]
NICE

**Pull request merged**